hackett-lib/hackett/base.rkt

```diff
@@ -2,12 +2,14 @@
 
 (require (only-in hackett/private/adt case* case λ λ* lambda lambda* defn _)
          (only-in hackett/private/type-alias type)
+         (only-in hackett/private/pattern-alias pattern List)
```

Comment:
I think both of these should be documented, but otherwise I think this all looks good.

Reply:
Given that we aren’t exporting `prop:case-pattern-expander`, it seems safe to document `pattern` as if that doesn’t exist.

Reply:
I'm having trouble with writing Hackett examples in the documentation. I'm trying to write function that uses the pattern aliases `Var`, `App`, and `Lam` like this:
```racket
(pattern (Var x)   (E (Var* x)))
(pattern (App a b) (E (App* a b)))
(pattern (Lam x a) (E (Lam* x a)))

(defn free : {Exp t:-> (t:List t:String)}
    [[(Var x)] (List x)]
    [[(App f a)] {(free f) ++ (free a)}]
    [[(Lam x b)] (filter (/= x) (free b))])
```
But the `hackett-examples` form is giving me error messages like this:
```
..../scribble-lib/scribble/eval.rkt:308:23: examples: exception raised in example
  error: "free: no inferred type"
```

Do you know why it isn't registering that `free` is defined with a type?

Reply:
Fixed by https://github.com/lexi-lambda/hackett/commit/76821cbc94b7cf9482c6fcab28746c54fc8f8a2e.